### PR TITLE
feat: add ability to inspect a specific shared worker

### PR DESCRIPTION
**Pull request by JohnPop117**
#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes:

### CODE REVIEW

File: docs/api/web-contents.md [hunk 1]
```diff
@@ -1463,6 +1463,16 @@ Starts inspecting element at position (`x`, `y`).
 
 Opens the developer tools for the shared worker context.
 
+#### `contents.inspectSharedWorker()`
+
+Opens the developer tools for the shared worker context.
```
Comment (alexeykuzmin) [inline docs/api/web-contents.md]: This doesn't look correct.

File: shell/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -2453,6 +2501,9 @@ void WebContents::BuildPrototype(v8::Isolate* isolate,
       .SetMethod("getOwnerBrowserWindow", &WebContents::GetOwnerBrowserWindow)
       .SetMethod("inspectServiceWorker", &WebContents::InspectServiceWorker)
       .SetMethod("inspectSharedWorker", &WebContents::InspectSharedWorker)
+      .SetMethod("inspectSharedWorkerById",
+                 &WebContents::InspectSharedWorkerById)
+      .SetMethod("getAllSharedWorkers", &WebContents::GetAllSharedWorkers)
```
Comment (alexeykuzmin) [inline shell/browser/api/atom_api_web_contents.cc]: Is not described in the docs.

File: spec/api-web-contents-spec.js [hunk 3]
```diff
@@ -1273,4 +1273,25 @@ describe('webContents module', () => {
       expect(data).to.be.an.instanceof(Buffer).that.is.not.empty()
     })
   })
+
+  // FIXME: disable during chromium update due to crash in content::WorkerScriptFetchInitation::CreateScriptLoaderOnIO
+  xdescribe('Shared Workers', () => {
+    it('can get multiple shared workers', async () => {
+      await w.loadFile(path.join(fixtures, 'api', 'shared-worker', 'shared-worker.html'))
+      await new Promise(resolve => setTimeout(resolve, 1000))
```
Comment (alexeykuzmin) [inline spec/api-web-contents-spec.js]: What for?

File: shell/browser/api/atom_api_web_contents.cc [hunk 4]
```diff
@@ -1527,7 +1539,43 @@ void WebContents::InspectElement(int x, int y) {
     OpenDevTools(nullptr);
   managed_web_contents()->InspectElement(x, y);
 }
+void WebContents::InspectSharedWorkerById(std::string workerId) {
```
Comment (miniak) [inline shell/browser/api/atom_api_web_contents.cc]: `const std::string&`

File: docs/api/web-contents.md [hunk 5]
```diff
@@ -1463,6 +1463,16 @@ Starts inspecting element at position (`x`, `y`).
 
 Opens the developer tools for the shared worker context.
 
+#### `contents.inspectSharedWorkerById(workerId)`
+
+* `workerId` String
+
+Inspects the shared worker based on it's id
+
+#### `contents.getAllSharedWorkers()`
```
[the remark this answers is not in the document]
Comment (miniak) [reply docs/api/web-contents.md]: please use `npm run create-typescript-definitions` and check the generated `electron.d.ts`

**Comment by alexeykuzmin** [comment]
> release-notes — Missing release notes

Please add release notes. It is definitely needed for a new API.

File: spec/api-web-contents-spec.js [hunk 6]
```diff
@@ -1273,4 +1273,29 @@ describe('webContents module', () => {
...
+      worker1.port.start()
+      worker2.port.start()
+      const contents = webContents.getAllWebContents()
+      const vectorOfWorkers = contents[0].getAllSharedWorkers()
+
+      expect(vectorOfWorkers.length).to.equal(2)
+      expect(vectorOfWorkers[0].url).to.contain('shared-worker')
+      expect(vectorOfWorkers[1].url).to.contain('shared-worker')
+
+      contents[0].inspectSharedWorkerById(vectorOfWorkers[0].id)
```
Comment (alexeykuzmin) [inline spec/api-web-contents-spec.js]: Please use two tests for two new methods.

File: shell/browser/api/atom_api_web_contents.cc (hunk 2 above)
Comment (alexeykuzmin) [reply shell/browser/api/atom_api_web_contents.cc]: Fixed.

File: docs/api/web-contents.md [hunk 7]
```diff
@@ -1463,6 +1463,16 @@ Starts inspecting element at position (`x`, `y`).
 
 Opens the developer tools for the shared worker context.
 
+#### `contents.inspectSharedWorkerById(workerId)`
+
+* `workerId` String
+
+Inspects the shared worker based on it's id
```
Comment (alexeykuzmin) [inline docs/api/web-contents.md]: Please use full stops at the end of sentences.

File: spec/api-web-contents-spec.js (hunk 6 above)
Comment (JohnPop117) [reply spec/api-web-contents-spec.js]: Fixed

File: docs/api/structures/shared-worker-info.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# WorkerInfo Object
+
+* `id` String - The unique id of the shared worker
```
Comment (malept) [inline docs/api/structures/shared-worker-info.md]:
```suggestion
* `id` String - The unique ID of the shared worker
```

File: docs/api/structures/shared-worker-info.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
...
+* `url` String - The url of the shared worker
```
Comment (malept) [inline docs/api/structures/shared-worker-info.md]:
```suggestion
* `url` String - The URL of the shared worker
```

File: docs/api/web-contents.md [hunk 10]
```diff
@@ -1463,6 +1463,16 @@ Starts inspecting element at position (`x`, `y`).
 
 Opens the developer tools for the shared worker context.
 
+#### `contents.inspectSharedWorkerById(workerId)`
+
+* `workerId` String
+
+Inspects the shared worker based on it's id.
```
Comment (malept) [inline docs/api/web-contents.md]:
```suggestion
Inspects the shared worker based on its ID.
```

File: docs/api/web-contents.md [hunk 11]
```diff
@@ -1463,6 +1463,16 @@ Starts inspecting element at position (`x`, `y`).
...
+
+#### `contents.getAllSharedWorkers()`
+
+Returns `WorkerInfo[]` - Vector of WorkerInfo objects of sharedworkers.
```
Comment (malept) [inline docs/api/web-contents.md]:
```suggestion
Returns `WorkerInfo[]` - Array of WorkerInfo objects of shared workers.
```

**Comment by MarshallOfSound** [comment]
Something has gone terribly wrong here, there are like 144 commits in this PR 😆

**Comment by JohnPop117** [comment]
Yes, my rebase seems to have failed spectacularly.

**Comment by alexeykuzmin** [comment]
Closed in favour of #19793.

**Pull request closed without merging**